octoprint_mrbeam/software_update_information.py

```diff
@@ -100,231 +215,244 @@ def switch_software_channel(plugin, channel):
-        pip_command=pip_command,
-        displayName="mrb_hw_info",
-        type="bitbucket_commit",
-        repo="mrb_hw_info",
-        api_user="MrBeamDev",
-        api_password="v2T5pFkmdgDqbFBJAqrt",
-        pip="git+ssh://git@bitbucket.org/mrbeam/mrb_hw_info.git@{target_version}",
-    )
-
-
-@logExceptions
-def _get_octo_plugin_description(module_id, tier, plugin, **kwargs):
-    """Additionally get the version from plugin manager (doesn't it do that by default??)"""
-    # Commented pluginInfo -> If the module is not installed, then it Should be.
-    pluginInfo = plugin._plugin_manager.get_plugin_info(module_id)
-    # if pluginInfo is None:
-    #     return {}
-    if tier == SW_UPDATE_TIER_DEV:
-        # Fix: the develop branches are not formatted as "mrbeam2-{tier}"
-        _b = DEFAULT_REPO_BRANCH_ID[SW_UPDATE_TIER_DEV]
-        kwargs.update(branch=_b, branch_default=_b)
-    return _get_package_description(
-        module_id=module_id, tier=tier, displayVersion=pluginInfo.version, **kwargs
-    )
 
+        fileversion = update_info.pop("version")
+        _logger.info("Software update file version: %s", fileversion)
+        defaultsettings = update_info.pop("default")
 
-@logExceptions
-def _get_package_description_with_version(
-    module_id, tier, package_name, pip_command, **kwargs
-):
-    """Additionally get the version diplayed through pip_command"""
-    if tier == SW_UPDATE_TIER_DEV:
-        # Fix: the develop branches are not formatted as "mrbeam2-{tier}"
-        _b = DEFAULT_REPO_BRANCH_ID[SW_UPDATE_TIER_DEV]
-        kwargs.update(branch=_b, branch_default=_b)
-
-    version = get_version_of_pip_module(package_name, pip_command)
-    if version:
-        kwargs.update(dict(displayVersion=version))
-
-    return _get_package_description(
-        module_id=module_id, tier=tier, pip_command=pip_command, **kwargs
-    )
-
-
-def _get_package_description(
-    module_id,
-    tier,
-    displayName=None,
-    displayVersion=None,
-    type="github_commit",
-    user="mrbeam",
-    repo=None,
-    branch="mrbeam2-{tier}",
-    branch_default="mrbeam2-{tier}",
-    restart="environment",
-    **kwargs
-):
-    """Shorthand to create repo details for octoprint software update plugin to handle."""
-    displayName = displayName or module_id
-    if "{tier}" in branch:
-        branch = branch.format(tier=get_tier_by_id(tier))
-    if "{tier}" in branch_default:
-        branch_default = branch_default.format(tier=get_tier_by_id(tier))
-    update_info = dict(
-        tier=tier,
-        displayName=displayName,
-        displayVersion=displayVersion,
-        user=user,
-        type=type,
-        repo=repo,
-        branch=branch,
-        branch_default=branch_default,
-        restart=restart,
-        **kwargs
-    )
-    return {module_id: update_info}
+        try:
+            _set_octoprint_config(
+                plugin, tier, update_info.pop("octoprint"), beamos_date
+            )
```

Comment:
Is there a reason why you want to `.pop()` those values instead of `.get()` ? Couldn't it lead to unpredicted missing info later on?

Reply:
I pop it so it wont be handeld by the for loop that processes all the module infos

Reply:
In that case I would say that they shouldn't be linked to the same node if they are not going to be treated the same; so we don't have to do risky things like this.

Reply:
Ok I can put it in a separate node, but why it is risky to pop it? I know that I don't need this data later